apptools/preferences/preferences.py

```diff
@@ -541,6 +541,14 @@ def _remove_preferences_listener(self, listener):
 
     def _set(self, key, value):
         """ Set the value of a preference in this node. """
+        
+        # everything must be unicode encoded so that ConfigObj configuration
+        # can properly serialize the data. If we receive regular str, they are
+        # supposed to be utf-8 encoded
+        if isinstance(value, str):
+            value = value.decode('utf-8')
```

Comment:
I think that if we get a `str` from the Python side, we should assume it to be ASCII, not UTF-8. We should not be throwing around UTF-8-encoded `str` objects in our Python code.